### ISSUE TITLE
[dev] [tooling] Provide python venv workflow for Coq python needs

### ISSUE THREAD

**Issue by ejgallego** [rocq-prover/rocq#17809]
Hi folks,

testing #17808 I did use a python `venv` setup, which works pretty well for people like me still not on Nix but on an APT-based stable distro. However I needed to make some adjustments.

I was wondering if we should indeed provide a developer `venv` target that would setup all python devs locally for devs, and also for the CI.

This way, we would not run `pip` in the Docker CI, but as a part of the build setup. WDYT?

It took very little time in my system to install sphinx in the venv (2 secs?)

**Comment by SkySkimmer** [comment]
>we would not run pip in the Docker CI

Eventually we will need to use some venv thing in the CI too as running pip without it got disabled recently

I would certainly appreciate a dev target as I'm pretty lost with this python stuff.

**Comment by ejgallego** [comment]
Yes, I'm quite lost too, for example I had python 3.10 which is not compatible with the stuff we do in docker-pip, you quickly run into a hell (and it seems we are far from alone on this)

I'll try to at least document my setup.

**Comment by Zimmi48** [comment]
Unfortunately, venv has some limitations, including not being able to specify the Python version you use. However, providing a `requirements.txt` or equivalent and instructions to generate a venv from it should still probably be useful for reproducibility of build setups for anyone who doesn't use Nix yet.
